### PR TITLE
cawbird: init at 1.0.1, replace corebird

### PR DESCRIPTION
**Pull request by schmittlauch**
###### Motivation for this change

Cawbird is a fork of the discontinued Corebird Twitter client.

We probably want to drop Corebird from nixpkgs as it doesn't even work anymore. The question is whether we want to make the `corebird` attribute point to `cawbird` instead for easy transitioning.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @jonafato

### CODE REVIEW

File: pkgs/applications/networking/cawbird/default.nix
Comment (jonringer) [inline pkgs/applications/networking/cawbird/default.nix]: personal preference
```suggestion
  meta = with stdenv.lib; {
```

File: pkgs/applications/networking/cawbird/default.nix
Comment (jonringer) [inline pkgs/applications/networking/cawbird/default.nix]:
```suggestion
    license = licenses.gpl3;
```

File: pkgs/applications/networking/cawbird/default.nix
Comment (jonringer) [inline pkgs/applications/networking/cawbird/default.nix]:
```suggestion
    platforms = platforms.linux;
```

File: pkgs/applications/networking/cawbird/default.nix
Comment (jonringer) [inline pkgs/applications/networking/cawbird/default.nix]:
```suggestion
    maintainers = with maintainers; [ jonafato schmittlauch ];
```

Comment (jonringer) [inline pkgs/applications/networking/cawbird/default.nix]: should jonafato be here?

File: pkgs/applications/networking/cawbird/default.nix
Comment (jonringer) [inline pkgs/applications/networking/cawbird/default.nix]: you might be able to simplify this with a prePatch step:
```suggestion
  prePatch = ''
    chmod +x data/meson_post_install.py # patchShebangs requires executable file
  '';
```

File: pkgs/applications/networking/cawbird/default.nix
Comment (schmittlauch) [reply pkgs/applications/networking/cawbird/default.nix]: That's his choice. As I based this nix expression on the one for Corebird where he is the maintainer, I'd assume that he's also interested in maintaining the continuation fork.

On the other hand the project has been non-functional since the beginning of this year, so he might have lost interest.

@jonafato

File: pkgs/applications/networking/cawbird/default.nix
Comment (schmittlauch) [reply pkgs/applications/networking/cawbird/default.nix]: so patchShebangs is already part of the normal patchPhase?

File: pkgs/applications/networking/cawbird/default.nix
Comment (jonringer) [reply pkgs/applications/networking/cawbird/default.nix]: That's fair, not sure how often is package is likely to receive attention.

**Comment by schmittlauch** [comment]
@Lassulus I marked COrebird as broken and added a comment in the expression referencing cawbird.
Is there a fancier way of raising a warning?

File: pkgs/applications/networking/cawbird/default.nix
Comment (schmittlauch) [reply pkgs/applications/networking/cawbird/default.nix]: I leave @jonafato as a maintainer until he complains or removes himself.

File: pkgs/applications/networking/cawbird/default.nix
Comment (schmittlauch) [reply pkgs/applications/networking/cawbird/default.nix]: I checked, this simplification does not work.

**Comment by schmittlauch** [comment]
From my perspective this PR is ready, I just squashed all cawbird-related commits.

Sorry @FRidh, I don't need your review. The notification has been cause by an improper rebase.

**Comment by schmittlauch** [comment]
@GrahamcOfBorg build cawbird

File: pkgs/applications/networking/cawbird/default.nix
Comment (jonringer) [reply pkgs/applications/networking/cawbird/default.nix]: :( sorry mate

**Comment by Lassulus** [comment]
I think the sdCard commit is misplaced?

**Comment by schmittlauch** [comment]
@Lassulus yes it is, but I don't know how to get rid of it.

I'm surprised it was added to the PR diff, as I just picked the existing commits in their previous order in my rebase -_-

**Comment by schmittlauch** [comment]
Btw, this should also be backported to 19.09

**Comment by jonringer** [comment]
`git rebase -i HEAD~4`
which will take you into a menu with all of your commits, then just replace `pick` with ... whatever delete/drop/remove command is, and save

**Comment by schmittlauch** [comment]
@jonringer Fixed it. Still am surprised that just touching a commit without changing it with rebase sets it to "committed by me" :\

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

**Comment by Lassulus** [comment]
> @jonringer Fixed it. Still am surprised that just touching a commit without changing it with rebase sets it to "committed by me" :\

if you change the base, you change the commit hash and so you are partly responsible

**Comment by schmittlauch** [comment]
> if you change the base, you change the commit hash and so you are partly responsible

But I did not change the base but only the most recent commits, leaving their predecessors completely unchanged (`pick`).

**Comment by schmittlauch** [comment]
@jonringer all checks have passed

**Comment by jonringer** [comment]
code LGTM, I would just like another opinion about deprecation mechanisms. @Ma27 whats your opinion?

**Comment by Ma27** [comment]
In that case I guess that it's fine to use `cawbird` as upstream chose to use a different name. 

Regarding corebird: as it seems to be/get EOLed I'd vote for dropping the package entirely and add a note in the release notes (we could also do `corebird = throw "use cawbird instread";` for some time before dropping the attribute).

**Comment by Lassulus** [comment]
so we merge this now and @schmittlauch makes a new PR where he deletes corebird and throws the error message in all-packges.nix ?

**Comment by schmittlauch** [comment]
@Lassulus As this PR already marks Corebird as broken, I better drop Corebird right away in this PR. Give me a minute…

**Comment by schmittlauch** [comment]
@Lassulus done

**Comment by Lassulus** [comment]
huh github now shows the file as moved and shows the diff internally, a little bit confusing but I guess we can't do anything about it? I will try to rerun it and merge it afterwards

**Comment by schmittlauch** [comment]
If we want to get this into 19.09, shall I add an entry to the release notes in the backport?

**Comment by Lassulus** [comment]
yes, add the notes to the stable manual in the backport

**Review by Lassulus — APPROVED** [review]
tested with nix-review, opened it, watched my twitter timeline

**Comment by schmittlauch** [comment]
backport: #70287

**Comment by worldofpeace** [comment]
backported in 8349643 b919677 5e0b687

**Pull request merged**
